src/Workspaces/SharedUtilitiesAndExtensions/Workspace/VisualBasic/LanguageServices/VisualBasicRemoveUnnecessaryImportsService.vb

```diff
@@ -0,0 +1,54 @@
+﻿' Licensed to the .NET Foundation under one or more agreements.
+' The .NET Foundation licenses this file to you under the MIT license.
+' See the LICENSE file in the project root for more information.
+
+Imports System.Composition
+Imports System.Threading
+Imports Microsoft.CodeAnalysis
+Imports Microsoft.CodeAnalysis.Formatting
+Imports Microsoft.CodeAnalysis.Host.Mef
+Imports Microsoft.CodeAnalysis.Internal.Log
+Imports Microsoft.CodeAnalysis.RemoveUnnecessaryImports
+Imports Microsoft.CodeAnalysis.VisualBasic.Syntax
+
+Namespace Microsoft.CodeAnalysis.VisualBasic.RemoveUnnecessaryImports
+    <ExportLanguageService(GetType(IRemoveUnnecessaryImportsService), LanguageNames.VisualBasic), [Shared]>
+    Partial Friend NotInheritable Class VisualBasicRemoveUnnecessaryImportsService
```

Comment:
so the entire structure of hte unnecessary imports stuff changed.  can we extract that into an independent change so that it's clear what's going on?  this is really interspersed with a ton of other stuff and i have to admit i don't quite get what's going on here and why it was necessary to change.

Reply:
Let me create a separate PR for unnecessary imports changes.

Reply:
I'll wait for https://github.com/dotnet/roslyn/pull/41550 to go in to avoid merge conflicts.

Reply:
>Let me create a separate PR for unnecessary imports changes.

👍 

Note: it may not be necessary to make any changes here at all if my PR goes in.  At that point SyntaxGenerator will simply exist in the codestyle layer and can be used directly.

Reply:
This is not going to be possible as this depends on syntax facts service also moving to shared layer (and all the extension methods used by these service), which is done in this PR. I will address any feedback on unnecessary imports refactoring in this PR. Let me provide a brief comment on changes that were done here:

1. Currently we have 2 language services: [IUnnecessaryImportsService](http://sourceroslyn.io/#Microsoft.CodeAnalysis.Features/RemoveUnnecessaryImports/IUnnecessaryImportsService.cs,73570cdc68553344) to compute unnecessary imports  and [IRemoveUnnecessaryImportsService](http://sourceroslyn.io/#Microsoft.CodeAnalysis.Features/RemoveUnnecessaryImports/IRemoveUnnecessaryImportsService.cs,b9e9baa8a091252f) to remove unnecessary imports from the given document.
2. This PR changes `IUnnecessaryImportsService` to `IUnnecessaryImportsProvider`, which provides the exact same functionality, but not as a language service. This basically enables us to switch [AbstractRemoveUnnecessaryImportsDiagnosticAnalyzer](http://sourceroslyn.io/#Microsoft.CodeAnalysis.Features/RemoveUnnecessaryImports/AbstractRemoveUnnecessaryImportsDiagnosticAnalyzer.cs,119) to be moved to the CodeStyle layer based on this provider.
3. This PR also changes the implementation of `IRemoveUnnecessaryImportsService` to use this provider to compute unnecessary imports and only contain the removal logic instead of having [AbstractRemoveUnnecessaryImportsService](http://sourceroslyn.io/#Microsoft.CodeAnalysis.Features/RemoveUnnecessaryImports/AbstractRemoveUnnecessaryImportsService.cs,1a00d47c7e649b76) implement both logic to compute unnecessary imports and also removal of them.